### DISCUSSION TITLE
Couch coop?

### DISCUSSION THREAD

**Discussion by vlolteanu** [diasurgical/DevilutionX#1745] [Q&A] [answered]
Hi,

Are there any plans to merge the code from the PS port that allows for local coop?

My gut tells me that this won't be trivial and will involve huge changes to the UI and net code. The latter could probably be solved by:
 * completely disabling online multiplayer at compile time when building for local multiplayer, or
 * feigning the presence of an extra networked client (probably extremely hard to pull off unless limiting couch coop to the machine hosting the game),

**Comment by AJenbo** [comment, marked as answer]
The way this was done on the PS port was invasive and stripped the support for network multiplayer. However, @yuripourre is working on a couch coop implementation.

**Reply by yuripourre** [reply]
Yep, as @AJenbo mentioned I have plans to do so but don't expect it any time soon. It requires a lot of changes in the UI, game logics and controls. And each step is reviewed and usually not welcome at the first time so it will require a looong time to be ready.

**Reply by AJenbo** [reply]
Sorry if you feel it's not welcome, it defiantly is (I'm pretty excited to have this feature comming) we just want it to be safely introduced :)
